test/ngResource/resourceSpec.js

```diff
@@ -1357,4 +1356,79 @@ describe('resource', function() {
   });
 
 
+  it('should be able to get meta data from query response', function() {
+    var result = [
+      { thing: 'value1' },
+      { thing: 'value2' }
+    ];
+    result.meta = 'data';
+
+    $httpBackend.expectGET('URL').respond({});
+    var models = $resource('URL', { }, {
+      query: {
+        method: 'GET',
+        isArray: true,
+        transformResponse: function() {
+          return result;
+        }
+      }
+    }).query();
+
+    $httpBackend.flush();
+
+    // Should have array and metadata
+    expect(models.length).toBe(2);
+    expect(models[0].thing).toEqual(result[0].thing);
+    expect(models[1].thing).toEqual(result[1].thing);
+    expect(models.$responseData.meta).toBe('data');
+  });
+
+
+  it('should be able to get error data from save response', function() {
+    var model = {
+      untouched: true,
+      error: 'No Error'
+    };
+
+    model = new ($resource('URL', { }, {
+      save: {
+        method: 'POST'
+      }
+    }))(model);
+
+    // Should have the config we passed the constructor
+    expect(model.untouched).toBe(true);
+    expect(model.error).toBe('No Error');
+
+    $httpBackend.expectPOST('URL').respond(500, { error: 'An error occurred' });
+    model.$save();
+    $httpBackend.flush();
+
+    // Should have the same config as before; decorated with props from response
+    expect(model.untouched).toBe(true);
```

Comment:
It might be worth verifying that `model.error` hasn't been updated, since the response contains the `error` property.